src/irr/builtin/CMakeLists.txt

```diff
@@ -1,19 +1,33 @@
-cmake_minimum_required(VERSION 3.9)
+function(irr_add_builtin_resource)
 
-function(irr_add_resource filename variable)
-  message(STATUS "Starting generating resource from ${filename}")
+  # generate header file
   add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/build/${filename}.h
-    COMMAND "${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/embedResource.py ${CMAKE_CURRENT_SOURCE_DIR}/${filename} ${variable} ${CMAKE_CURRENT_BINARY_DIR}/build/${filename}.h"
-    COMMENT "Generated resource from ${filename}"
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResources.h
+    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/buitlinHeaderGen.py ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResources.h ${CMAKE_SOURCE_DIR} ${BUILTIN_RESOURCE_LIST}
+    COMMENT "Generated built-in resource headers"
     VERBATIM
   )
-  add_custom_target(builtin_resource DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/build/${filename}.h)
 
-  # execute_process(COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/embedResource.py ${CMAKE_CURRENT_SOURCE_DIR}/${filename})
+  # generate cpp file with binary data of resources inside
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResourceData.cpp
+    COMMAND ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/builtinDataGen.py ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResourceData.cpp ${CMAKE_SOURCE_DIR} ${BUILTIN_RESOURCE_LIST}
+    COMMENT "Generated built-in resource binary data"
+    VERBATIM
+  )
+
+  # set up dependency
+  add_custom_target(builtin_resources DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResourceData.cpp ${CMAKE_CURRENT_BINARY_DIR}/build/builtinResources.h)
+
+  # for testing purposes
+  # execute_process(COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/builtinHeaderGen.py ${CMAKE_CURRENT_SOURCE_DIR}/builtinResources.h ${CMAKE_SOURCE_DIR} ${BUILTIN_RESOURCE_LIST})
+  # execute_process(COMMAND python ${CMAKE_CURRENT_SOURCE_DIR}/builtinDataGen.py ${CMAKE_CURRENT_SOURCE_DIR}/builtinResourceData.cpp ${CMAKE_SOURCE_DIR} ${BUILTIN_RESOURCE_LIST})
 endfunction()
 
 
 if(IRR_EMBED_BUILTIN_RESOURCES)
-	irr_add_resource(shader1.txt embeddedShader)
+  #fill this list with resource paths relative to top level of IrrlichtBaW's source
+  set(BUILTIN_RESOURCE_LIST src/irr/builtin/shader1.txt)
+
+	irr_add_builtin_resource()
```

Comment:
this has really bad syntax

Reply:
you should be passing the list to the function, not using a weird global cmake variable